src/modules/ui/components/layout/index.js

```diff
@@ -86,6 +86,13 @@ class Layout extends React.Component {
       downPanelDefaultSize = downPanelInRight ? 400 : 200;
     }
 
+    if(typeof localStorage !== 'undefined') {
```

Comment:
This is not enough to make sure localStorage is usable. When using Safari in private mode, you'll run into issues because localStorage is defined, but the quota is set to 0, meaning that if you try to setItem, the browser will throw an error.

Pretty simple to get around though, could do something like this:

```javascript
const hasLocalStorage = () => {
  try {
      localStorage.setItem('yo', 'yo')
      localStorage.removeItem('yo')
      return true
  } catch(e) {
      return false
  }
}
```